### PR TITLE
Acknowledge serialization options of CosmosClient for PatchOperationBuilder

### LINKED ISSUE #270: Patch Update not working when Serialization Options Property Naming Policy is set to Default.
**Describe the bug**
When "propertynamingpolicy" in set to "Default" in SerializationOptions the SDK throws Bad Request Exception since its trying to find a property and could nt find. Sdk is not considering Serialization settings for PropertyNamingPolicy when creating PatchOperations.

Example :
When configuration is set to "Default". Property name is "Name".
But when trying to replace the value at "Name" property is searching for "name".

**To Reproduce**
Set "PropertyNamingPolicy" under "SerializationOptions" to Default.
image

Perform patch update using UpdateAsync
image

**Expected behavior**
Successfully perform Patch update on document,

**Actual behavior**
SDK throws 400 Bad request execption.

**Environment summary**
SDK Version: 3.5.3
OS Version (Windows)

### PR DESCRIPTION
**Pull request by MO2k4**
This pr acknowledge the used serialization option of the cosmos client for the patch operation builder

- fixes #270

### CODE REVIEW

**Comment by IEvangelist** [comment]
@all-contributors please add @MO2k4 for code.

File: src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs [hunk 1]
```diff
@@ -17,12 +17,24 @@ namespace Microsoft.Azure.CosmosRepository.Builders
...
+        private readonly NamingStrategy _namingStrategy;
 
         internal readonly List<InternalPatchOperation> _rawPatchOperations = new();
 
         public IReadOnlyList<PatchOperation> PatchOperations => _patchOperations;
 
+        public PatchOperationBuilder()
+        {
+            _namingStrategy = new CamelCaseNamingStrategy();
+        }
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs:29]:
```suggestion
        public PatchOperationBuilder() =>
            _namingStrategy = new CamelCaseNamingStrategy();
```

File: src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs [hunk 2]
```diff
@@ -17,12 +17,24 @@ namespace Microsoft.Azure.CosmosRepository.Builders
...
+
+        public PatchOperationBuilder(CosmosPropertyNamingPolicy? cosmosPropertyNamingPolicy)
+        {
+            _namingStrategy = cosmosPropertyNamingPolicy == CosmosPropertyNamingPolicy.Default
+                ? new DefaultNamingStrategy()
+                : new CamelCaseNamingStrategy();
+        }
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs:36]:
```suggestion
        public PatchOperationBuilder(CosmosPropertyNamingPolicy? cosmosPropertyNamingPolicy) => 
            _namingStrategy = cosmosPropertyNamingPolicy == CosmosPropertyNamingPolicy.Default
                ? new DefaultNamingStrategy()
                : new CamelCaseNamingStrategy();
```

File: src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs [hunk 3]
```diff
@@ -32,24 +44,14 @@ public IPatchOperationBuilder<TItem> Replace<TValue>(Expression<Func<TItem, TVal
...
-                {
-                    return jsonAttribute.PropertyName;
-                }
-            }
-
-            return _namingStrategy.GetPropertyName(propertyInfo.Name, false);
+            JsonPropertyAttribute[] attributes =
+                propertyInfo.GetCustomAttributes<JsonPropertyAttribute>(true).ToArray();
+
+            return attributes.Length == 0
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs:52]:
```suggestion
            return attributes.Length is 0
```

File: src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs [hunk 4]
```diff
@@ -32,24 +44,14 @@ public IPatchOperationBuilder<TItem> Replace<TValue>(Expression<Func<TItem, TVal
...
+                ? _namingStrategy.GetPropertyName(propertyInfo.Name, false)
+                : attributes.First().PropertyName;
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Builders/PatchOperationBuilder.cs:54]:
```suggestion
                : attributes[0].PropertyName;
```

It's an array, use the indexer.

**Review by IEvangelist — COMMENTED** [review]
A few nits, but this looks really good.

**Comment by MO2k4** [comment]
Thanks for the comments. I will add the array usage to the test as well 🙂

**Review by MO2k4 — COMMENTED** [review]
Changed to use array indexer

**Review by mumby0168 — APPROVED** [review]

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**